### DISCUSSION TITLE
The power of Monaco Editor in OpenMage!

### DISCUSSION THREAD

**Discussion by empiricompany** [OpenMage/magento-lts#3665] [Show & Tell]
Hello, OpenMage community! 🚀

I'm excited to present to you a game-changing new extension [openmage-mm_monacoeditor](https://github.com/empiricompany/openmage-mm_monacoeditor) that integrates the powerful [Monaco Editor](https://github.com/microsoft/monaco-editor), utilized in Microsoft's VSCode, into OpenMage.

This extension replace textareas in CMS content and Design Update XML editing with Monaco - a feature-rich platform that gives your coding a real boost, while ensuring seamless indentation among a myriad of other exciting features.

Some of these splendid features include smart IntelliSense, syntax highlighting, code formatting/folding, and multiple cursors/selections. 

Plus, the cherry on the cake - it comes packed with Emmet, for speedy HTML and CSS coding, and Tailwindcss Intellisense, providing intelligent auto-completion, suggestions, and tooltips for Tailwind CSS classes!

Additionally, you can disable the TinyMCE WYSIWYG editor for specific static blocks and pages. This option has been added because TinyMCE removes code that does not conform to the HTML5 schema, such as Alpine.js.

![demo-monacoeditor](https://github.com/empiricompany/openmage-mm_monacoeditor/assets/5071467/05398366-fe69-498c-99f3-6140571bc395)
![configs](https://github.com/empiricompany/openmage-mm_monacoeditor/assets/5071467/f9ddde27-3fe9-4f10-889c-13e96c7fb5aa)

Check it out! Hope you find it useful!

**Comment by fballiano** [comment]
😱

**Reply by fballiano** [reply]
added to awesome-openmage

**Comment by ma4nn** [comment]
very cool idea 😍

**Comment by elidrissidev** [comment]
Very nice, I always thought code inputs should be implemented with something more convenient than just plain `textarea`s. Can you switch the editor theme to the lighter version so it matches the rest of the admin area?

**Reply by empiricompany** [reply]
theme is based on your system preferences PRs  are welcome
